### PR TITLE
AO3-4906 AO3-4828 Implements strong params for Work and ExternalWork

### PR DESCRIPTION
**Pull request by davidstump**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [x] Have you added tests for any changed functionality?
* [x] Have you added the JIRA issue number as the *first* thing in your pull request title (eg: `AO3-1234 Fix thing`)
* [x] Have you updated the JIRA issue with the information below?

## Issue

https://otwarchive.atlassian.net/browse/AO3-4828

## Purpose

Implements strong parameters and ForbiddenAttributesProtection for ExternalWork

## Testing

1. Log in as admin
2. Find a bookmark of an external work – http://test.ao3.org/external_works is giving a 500 error, but you can likely find one on http://test.ao3.org/bookmarks or http://test.ao3.org/users/testy/bookmarks
3. Follow the "Edit External Work" link
4. Change as much or as little information in the form as you want
5. Press "Update External Work"

## References

None

## Credit

Littlelines - David Stump

*What pronouns do you prefer (she/her, he/him, zie/hir etc)?*

he/him

### CODE REVIEW

File: app/controllers/bookmarks_controller.rb [hunk 1]
```diff
@@ -289,4 +289,17 @@ def index_page_title
...
   end
 
+  private
+
+  def bookmark_params
+    params.require(:bookmark).permit(
+      :bookmarkable_id, :bookmarkable_type,
+      :pseud_id, :notes, :tag_string, :collection_names, :private, :rec,
+      external: [
+        :url, :author, :title, :fandom_string, :rating_string, :relationship_string,
```
Comment (sarken) [inline app/controllers/bookmarks_controller.rb]: This might also need `category_string` -- I've added a test for it in #2797

Comment (davidstump) [inline app/controllers/bookmarks_controller.rb]: Done. Thanks :)

**Comment by davidstump** [comment]
Should be all set now!

**Comment by zz9pzza** [comment]
@davidstump this one has a merge conflct as well.

**Comment by davidstump** [comment]
Heads up - the test failures in this PR are a result of merging in master to fix merge conflicts. The errors are handled in a separate PR #2840.

**Pull request merged**